frontend/src/scenes/hog-functions/configuration/hogFunctionConfigurationLogic.tsx

```diff
@@ -619,7 +619,24 @@ export const hogFunctionConfigurationLogic = kea<hogFunctionConfigurationLogicTy
                 if (!values.hasAddon && values.type !== 'transformation') {
                     // Remove the source field if the user doesn't have the addon (except for transformations)
                     delete payload.hog
-                    delete payload.inputs_schema
+
+                    // Only preserve templating settings in inputs_schema, remove everything else
+                    if (payload.inputs_schema) {
+                        const modifiedSchema = payload.inputs_schema as any[] // Store reference before overwriting
+                        const originalSchema =
+                            values.hogFunction?.template?.inputs_schema || values.template?.inputs_schema || []
+
+                        payload.inputs_schema = originalSchema.map((originalInput) => {
```

Comment:
No way you could have known this but as a general rule - don't modify the schema ever.